### PR TITLE
Update leiden future warning

### LINKED ISSUE #2935: `FutureWarning` for the 1.11 to begin the process of switching leiden backend
See: https://github.com/scverse/scanpy/issues/2865

### PR DESCRIPTION
**Pull request by ivirshup**
- [x] Closes #2935
- [ ] Tests included or not required because:

- [x] Release notes not necessary because: Code has not been in a release

### CODE REVIEW

File: scanpy/_utils/__init__.py [hunk 1]
```diff
@@ -857,6 +858,13 @@ def warn_with_traceback(message, category, filename, lineno, file=None, line=Non
     settings.write(warnings.formatwarning(message, category, filename, lineno, line))
 
 
+def warn_once(msg: str, category: type[Warning], stacklevel: int = 1):
+    warnings.warn(msg, category, stacklevel=stacklevel)
+    # Prevent from showing up every time an awkward array is used
+    # You'd think `'once'` works, but it doesn't at the repl and in notebooks
```
Comment (ilan-gold) [inline scanpy/_utils/__init__.py]: What is this about awkward arrays?

Comment (ivirshup) [reply scanpy/_utils/__init__.py]:
```suggestion
    # You'd think `'once'` works, but it doesn't at the repl and in notebooks
```

Cause this is copied from anndata

File: scanpy/_utils/__init__.py [hunk 2]
```diff
@@ -857,6 +858,12 @@ def warn_with_traceback(message, category, filename, lineno, file=None, line=Non
     settings.write(warnings.formatwarning(message, category, filename, lineno, line))
 
 
+def warn_once(msg: str, category: type[Warning], stacklevel: int = 1):
+    warnings.warn(msg, category, stacklevel=stacklevel)
+    # You'd think `'once'` works, but it doesn't at the repl and in notebooks
```
Comment (flying-sheep) [inline scanpy/_utils/__init__.py:863]: can you link the bug report?

Comment (ivirshup) [reply scanpy/_utils/__init__.py:863]: For the bug this PR is fixing, or what?

Comment (flying-sheep) [reply scanpy/_utils/__init__.py:863]: no, in JupyterLab about it not working

**Pull request merged**